rmf_task/src/rmf_task/BackupFileManager.cpp

```diff
@@ -59,7 +62,17 @@ class BackupFileManager::Group::Implementation
   : group_directory(std::move(directory)),
     settings(std::move(settings))
   {
-    // Do nothing
+    std::filesystem::create_directories(this->group_directory);
+
+    // BH: Should we handle permission exceptions to give the user more info?
+    std::filesystem::permissions(this->group_directory,
```

Comment:
Linux system administration is not something I have much expertise in, but is it reasonable for us to be modifying directory permissions? My instinct is that this is something that should be managed outside of any RMF process, and any files or directories that we create should just use default permissions (which I assume is probably `owner_read | owner_write`?). I think we should especially stay away from any `_exec` permissions since these files are not meant to be executed.

But if you had a particular concern in mind when writing this, do let me know.

Reply:
perhaps we can recursively check permissions for folders / files we need instead of modifying them? It is possible to have a the group folder be accessible but files inside to be restricted. this seems to only check the group directory, but not its contents

Reply:
As a side note, I recommend wrapping the permission checking code into a simple function that takes in a `std::filesystem::path` and returns a boolean.